diff --git a/src/Constants.hpp b/src/Constants.hpp
index 92f37f0..2bac198 100644
--- a/src/Constants.hpp
+++ b/src/Constants.hpp
@@ -46,6 +46,9 @@ constexpr int k_rightDriveSlaveID = 14;
 // DriveTrain distance per pulse
 constexpr double k_driveDpP = 240.0 / ((6552 + 6522) / 2.0);  // in/pulse

+// Robot width
+constexpr double k_robotWidth = 36.0;  // inches
+
 // DriveTrain position PID, Extra //'s mean practice PID values
 constexpr double k_driveMaxSpeed = 24000;  // in/sec
 constexpr double k_posP = 0.07;            // 0.07
diff --git a/src/Subsystems/DriveTrain.cpp b/src/Subsystems/DriveTrain.cpp
index f760824..bb39740 100644
--- a/src/Subsystems/DriveTrain.cpp
+++ b/src/Subsystems/DriveTrain.cpp
@@ -188,7 +188,10 @@ double DriveTrain::GetPosReference() const { return m_posRef.Get(); }

 double DriveTrain::GetAngleReference() const { return m_angleRef.Get(); }

-double DriveTrain::GetAngle() const { return m_gyro.GetAngle(); }
+double DriveTrain::GetAngle() const {
+  return 180.0 * (m_leftGrbx.GetPosition() - m_rightGrbx.GetPosition()) /
+         (k_robotWidth * M_PI);
+}

 double DriveTrain::GetRate() const { return m_gyro.GetRate(); }

diff --git a/src/Subsystems/DriveTrain.hpp b/src/Subsystems/DriveTrain.hpp
index 3afd2c2..40c5bc3 100644
--- a/src/Subsystems/DriveTrain.hpp
+++ b/src/Subsystems/DriveTrain.hpp
@@ -96,7 +96,7 @@ private:

     // Angle PID
     ADXRS450_Gyro m_gyro;
-    FuncNode m_angleSensor{[&] { return m_gyro.GetAngle(); }};
+    FuncNode m_angleSensor{[this] { return GetAngle(); }};
     SumNode m_angleError{&m_angleRef, true, &m_angleSensor, false};
     PIDNode m_anglePID{k_angleP, k_angleI, k_angleD, &m_angleError};

